File: test/fixtures/node-axios/j_patch_array.js
import axios from 'axios';

const response = await axios.patch(
  'http://localhost:28139/patch',
  'item[]=1&item[]=2&item[]=3',
  {
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded'
    }
  }
);

File: test/fixtures/kotlin-ktor/j_patch_array.kt
import io.ktor.client.HttpClient
import io.ktor.client.request.header
import io.ktor.client.request.patch
import io.ktor.client.request.setBody
import io.ktor.client.statement.bodyAsText
import io.ktor.http.isSuccess
import kotlinx.coroutines.runBlocking

val client = HttpClient()

runBlocking {
    val response = client.patch("http://localhost:28139/patch") {
        setBody("item[]=1&item[]=2&item[]=3")
        header("Content-Type", "application/x-www-form-urlencoded")
    }
    if (!response.status.isSuccess()) throw Exception("Request failed with status code ${response.status}")

    println(response.bodyAsText())
}
